images/tlshd/src/cmd/main.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2024 Flant JSC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"bytes"
+	"context"
+	"flag"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"os/signal"
+	"regexp"
+	"syscall"
+	"time"
+)
+
+var (
+	opt = &Opt{}
+)
+
+type Opt struct {
+	TimeoutWait uint
+}
+
+func (o *Opt) Parse() {
+	flag.UintVar(&o.TimeoutWait, "timeout_wait", 0, "timeout in seconds for process execution (0 means no timeout)")
+	flag.Parse()
+}
+
+func main() {
+	opt.Parse()
+
+	args := []string{"-s"}
+	if opt.TimeoutWait > 0 {
+		args = append(args, "-c", "/opt/deckhouse/csi/etc/tlshd.conf")
+	} else {
+		args = append(args, "-c", "/etc/tlshd.conf")
+	}
+	cmd := exec.Command("/opt/deckhouse/csi/bin/tlshd", args...)
```

Comment:
Tehre's `exec.CommandContext`, which can be helpful, because you'd only need to cancel the context, without Killing the child process "manually"

Reply:
And then you could use `cmd.Output()`

Reply:
If contexts are used, it will only confuse the logic, because in our case, the termination of the child process due to a timeout is considered a positive outcome.